lib/representable/deserializer.rb

```diff
@@ -20,7 +20,13 @@ module Representable
   end
 
   OverwriteOnNil = ->(input, options) do
-    input.nil? ? (SetValue.(input, options); Pipeline::Stop) : input
+    if input.nil?
+      function = options[:binding][:setter] ? Setter : SetValue
```

Comment:
This is correct, but we can't inspect this at runtime for performance reasons, we have to maintain two `Override` functions and set them at compile time.